File: result/translate/chatgpt/java2cpp/c++/MINIMUM_PERIMETER_N_BLOCKS.cpp
long f_gold(int n) {
    int l = sqrt(n);
    int sq = l * l;
    if (sq == n) return l * 4;
    else {
        long row = n / l;
        long perimeter = 2 * (l + row);
        if (n % l != 0) perimeter += 2;
        return perimeter;
    }
}

File: result/translate/starchat/python2java/java/MINIMUM_PERIMETER_N_BLOCKS.java
public static int f_gold ( int n ) {
        int l = (int) Math.sqrt ( n );
        int sq = l * l;
        if ( sq == n ) return l * 4;
        int row = n / l;
        int perimeter = 2 * ( l + row );
        if ( n % l!= 0 ) perimeter += 2;
        return perimeter;
    }